DemoApp/StreamChat/StreamChatWrapper+DemoApp.swift

```diff
@@ -40,6 +40,7 @@ extension StreamChatWrapper {
         Components.default.messageListDateOverlayEnabled = true
         Components.default.isVoiceRecordingEnabled = true
         Components.default.messageSwipeToReplyEnabled = true
+        Components.default.channelListSearchFactory = ChannelListSearchStrategy.messages
```

Comment:
Should we try to have consistent naming here? We have Factory vs Strategy. (I like Strategy better, but your call)

Reply:
Yeah, I'm also not 100% ok with this. The thing is that this is actually a `Factory` since it is just a protocol that has 1 implementation to create a searchController. Our Search Strategy happens to conform to that protocol, and so it is also a factory 🤔 

I have 2 ideas:
1- Rename ChannelListSearchStrategy to `ChannelListSearchStrategyFactory` (Kinda weird)
2- Rename `channelListSearchFactory` to `channelListSearchStrategy` and remove the `ChannelListSearchFactory` protocol. Because, if we think about it, if you want to create the search controller from scratch, you don't need this protocol. You can just create it directly in the Channel List. So this factory protocol is a bit unnecessary now that I think about it 🤔

Reply:
Will go with option 2 then 👍